xbmc/music/tags/MusicInfoTagLoaderFFmpeg.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ *      Copyright (C) 2014 Arne Morten Kvarving
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "MusicInfoTagLoaderFFmpeg.h"
+#include "MusicInfoTag.h"
+#include "filesystem/File.h"
+#include "cores/FFmpeg.h"
+#include "utils/URIUtils.h"
+#include "utils/StringUtils.h"
+
+using namespace MUSIC_INFO;
+using namespace XFILE;
+
+static int vfs_file_read(void *h, uint8_t* buf, int size)
+{
+  CFile* pFile = static_cast<CFile*>(h);
+  return pFile->Read(buf, size);
+}
+
+static int64_t vfs_file_seek(void *h, int64_t pos, int whence)
+{
+  CFile* pFile = static_cast<CFile*>(h);
+  if (whence == AVSEEK_SIZE)
+    return pFile->GetLength();
+  else
+    return pFile->Seek(pos, whence & ~AVSEEK_FORCE);
+}
+
+CMusicInfoTagLoaderFFmpeg::CMusicInfoTagLoaderFFmpeg(void)
+{
+}
+
+CMusicInfoTagLoaderFFmpeg::~CMusicInfoTagLoaderFFmpeg()
+{
+}
+
+bool CMusicInfoTagLoaderFFmpeg::Load(const std::string& strFileName, CMusicInfoTag& tag, EmbeddedArt *art)
+{
+  tag.SetLoaded(false);
+
+  CFile file;
+  if (!file.Open(strFileName))
+    return false;
+
+  uint8_t* buffer = (uint8_t*)av_malloc(32768);
```

Comment:
Where does the magic number come from? Might be worth putting it in a define or const as you use it multiple times.